tests/test_trainer.py

```diff
@@ -411,5 +413,62 @@ class CurrentTestModel(
     trainer.test()
 
 
+def test_num_trainer_steps(tmpdir):
+    """Verify model trains according to speficied steps"""
+    tutils.reset_seed()
+    model, _ = tutils.get_model()
+
+    train_percent = 0.05
+    num_train_samples = math.floor(len(model.train_dataloader()) * train_percent)
+
+    trainer_options = dict(
+        gpus=None,
+        default_save_path=tmpdir,
+        train_percent_check=train_percent,
+    )
+
+    trainer_options['max_epochs'] = 5
+    trainer_options['max_steps'] = num_train_samples + 10
+    trainer = Trainer(**trainer_options)
+    result = trainer.fit(model)
+    assert result == 1
+    # should stop at max_steps
+    assert trainer.global_step == trainer_options['max_steps'], "Model did not stop at max_steps"
```

Comment:
could you please splot this test to multiple subtests so it is easier to address the issue if just one of them is failing...

Reply:
I split the test to min steps and max steps